05week/spaceTravelToMars.js

```diff
@@ -3,57 +3,124 @@
 let assert = require('assert');
 
 let jobTypes = {
-  pilot: 'MAV',
-  mechanic: 'Repair Ship',
-  commander: 'Main Ship',
-  programmer: 'Any Ship!'
+    pilot: 'MAV',
+    mechanic: 'Repair Ship',
+    commander: 'Main Ship',
+    programmer: 'Any Ship!'
 };
 
-// Your code here
+class CrewMember {
+    constructor(name, job, specialSkill, ship) {
+        this.name = name;
+        this.job = job;
+        this.specialSkill = specialSkill;
+        this.ship = null;
+    }
+    enterShip(assignShip) {
+        assignShip.crew.push(this);
+        this.ship = assignShip;
+    }
+}
+
+class Ship {
+    constructor(name, type, ability) {
+        this.name = name;
+        this.type = type;
+        this.ability = ability;
+        this.crew = [];
+    }
+    missionStatement() {
+        if (this.crew.length === 0) {
+            return "Can't perform a mission yet.";
```

Comment:
this does not check that there is a crew member who can perform the task.
i was looking for something like
```
    for (let i = 0; i < this.crew.length ; i++) {
        let job = this.crew[i].job;
        if ((this.type == jobTypes[job]) || (job == 'programmer')) {
          return this.ability;
        } 
      }
      return 'Can't perform a mission yet.";
```